climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1158 @@
+                    self.intensity = sparse.csr_matrix( \
+                        np.amax(self.intensity[evt_idx_merge], 0))
+                    self.event_id = np.array([np.max(self.event_id)+1])
+                    self.event_name = list(map(str, self.event_id))
+                    self.date = [date_start]
+                    self.date_end = [date_end]
+                    self.orig = np.ones(1, bool)
+                    self._set_frequency()
+                    self.fraction = self.intensity.copy()
+                    self.fraction.data.fill(1.0)
+                else:
+                    # merge event & append
+                    self.intensity = sparse.vstack([self.intensity, \
+                        np.amax(self.intensity[evt_idx_merge], 0)], format='csr')
+                    self.event_id = np.append(self.event_id, np.max(self.event_id)+1)
+                    self.event_name = list(map(str, self.event_id))
+                    self.date = np.append(self.date, date_start)
+                    self.date_end = np.append(self.date_end, date_end)
+                    self.orig = np.append(self.orig, np.ones(1, bool))
+
+                    # remove initial events
+                    self.intensity = self.intensity[np.delete(np.arange(0, \
+                        self.intensity.get_shape()[0]), evt_idx_merge)]
+                    self.event_id = np.delete(self.event_id, evt_idx_merge)
+                    self.event_name = list(map(str, self.event_id))
+                    self.date = np.delete(self.date, evt_idx_merge)
+                    self.date_end = np.delete(self.date_end, evt_idx_merge)
+                    self.orig = np.delete(self.orig, evt_idx_merge)
+                    self._set_frequency()
+                    self.fraction = self.intensity.copy()
+                    self.fraction.data.fill(1.0)
+
+            else:
+                self.intensity = sparse.csr_matrix(np.amax(self.intensity, 0))
+                self.event_id = np.array([np.max(self.event_id)+1])
+                self.event_name = list(map(str, self.event_id))
+                self.date = [np.min(self.date)]
+                self.date_end = [np.max(self.date_end)]
+                self.orig = np.ones(1, bool)
+                self._set_frequency()
+                self.fraction = self.intensity.copy()
+                self.fraction.data.fill(1.0)
+            LOGGER.info('The merged event has event_id %s', self.event_id[-1])
+
+        else:
+            self.intensity = sparse.csr_matrix(np.amax(self.intensity, 0))
+            self.event_id = np.array([np.max(self.event_id)+1])
+            self.orig = np.zeros(1, bool)
+            self._set_frequency()
+            self.fraction = self.intensity.copy()
+            self.fraction.data.fill(1.0)
+
+    def summarize_fires_to_seasons(self, year_start=None, year_end=None,
+                                   hemisphere=None):
+        """ Summarize historic fires into fire seasons.
```

Comment:
As in previous docstrings, could you add the method how you combine the fires? I.e. by taking the maximum intensity at each centroid.

Reply:
adjusted the docstring